drivers/adc/adc_nrfx_saadc.c

```diff
@@ -673,58 +629,48 @@ static int adc_nrfx_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
-static void saadc_irq_handler(const struct device *dev)
+static void event_handler(const nrfx_saadc_evt_t *event)
 {
-	if (nrf_saadc_event_check(NRF_SAADC, NRF_SAADC_EVENT_END)) {
-		nrf_saadc_event_clear(NRF_SAADC, NRF_SAADC_EVENT_END);
+	nrfx_err_t err;
 
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_STOP);
-
-#if defined(CONFIG_PM_DEVICE_RUNTIME)
-		pm_device_runtime_put(DEVICE_DT_INST_GET(0));
-#else
-		nrf_saadc_disable(NRF_SAADC);
-#endif
+	m_data.buffer = event->data.done.p_buffer;
```

Comment:
nit: this should be actually done inside `if (event->type == NRFX_SAADC_EVT_DONE)`